In [ ]:
Gerardo Carcoba - A01178753 - Carrera:LAF

# Comprensión de los Datos

In [48]:
#importa librerías
import pandas as pd

# Descripción de Variables


Pclass Passenger Class (1 = 1st; 2 = 2nd; 3 = 3rd): Categórica Nominal 
survival Survival (0 = No; 1 = Yes)  
name Name  
sex Sex  
age Age  
sibsp Number of Siblings/Spouses Aboard  
parch Number of Parents/Children Aboard  
ticket Ticket Number  
fare Passenger Fare (British pound)  
cabin Cabin  
embarked Port of Embarkation (C = Cherbourg; Q = Queenstown; S = Southampton)  
boat Lifeboat  
body Body Identification Number  
home.dest Home/Destination

**Ejemplo:** Crear un objeto DataFrame con base en un archivo .csv

In [49]:
#lee archivo csv
df = pd.read_csv("diabetes.csv")

In [50]:
#Usa función shape para revisar el total de renglones y columnas
df.shape

(768, 9)

In [51]:
#Revisa los primeros 5 renglones del dataset usando la función head()
df.head ()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [56]:
#Revisa los últimos 5 renglones del dataset usando la función tail()
df.tail()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0

In [57]:
#Revisa la información mas completa del conjunto de datos usando la función info()
#Muestra el total de datos, las columnas y su tipo correspondiente, dice si contiene nulos o no
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Pregnancies               768 non-null    int64  
 1   Glucose                   768 non-null    int64  
 2   BloodPressure             768 non-null    int64  
 3   SkinThickness             768 non-null    int64  
 4   Insulin                   768 non-null    int64  
 5   BMI                       768 non-null    float64
 6   DiabetesPedigreeFunction  768 non-null    float64
 7   Age                       768 non-null    int64  
 8   Outcome                   768 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


In [54]:
#revisa cuántos valores únicos tiene cada atributo del archivo usando la función nunique()
df.nunique()

Pregnancies                  17
Glucose                     136
BloodPressure                47
SkinThickness                51
Insulin                     186
BMI                         248
DiabetesPedigreeFunction    517
Age                          52
Outcome                       2
dtype: int64

## Exploración de Datos

In [26]:
#utiliza la función describe() para obtener estadística básica. se puede incluir -0
df.describe()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
count   768.000000  768.000000     768.000000     768.000000  768.000000   
mean      3.845052  120.894531      69.105469      20.536458   79.799479   
std       3.369578   31.972618      19.355807      15.952218  115.244002   
min       0.000000    0.000000       0.000000       0.000000    0.000000   
25%       1.000000   99.000000      62.000000       0.000000    0.000000   
50%       3.000000  117.000000      72.000000      23.000000   30.500000   
75%       6.000000  140.250000      80.000000      32.000000  127.250000   
max      17.000000  199.000000     122.000000      99.000000  846.000000   

              BMI  DiabetesPedigreeFunction         Age     Outcome  
count  768.000000                768.000000  768.000000  768.000000  
mean    31.992578                  0.471876   33.240885    0.348958  
std      7.884160                  0.331329   11.760232    0.476951  
min      0.000000                  0.078000   21.000000    0.000000  
25%     27.300000                  0.243750   24.000000    0.000000  
50%     32.000000                  0.372500   29.000000    0.000000  
75%     36.600000                  0.626250   41.000000    1.000000  
max     67.100000                  2.420000   81.000000    1.000000

In [ ]:
Los datos muestran una glucosa promedio de 120.9 y un BMI de 31.9, con alta variabilidad en insulina y grosor de piel, lo que indica diferencias marcadas en el perfil metabólico de los pacientes.

In [65]:
df["BMI"].describe()

count    768.000000
mean      31.992578
std        7.884160
min        0.000000
25%       27.300000
50%       32.000000
75%       36.600000
max       67.100000
Name: BMI, dtype: float64

In [ ]:
La mayoría de los participantes presenta sobrepeso u obesidad, lo que aumenta el riesgo metabólico.

In [66]:
df["Outcome"].describe()

count    768.000000
mean       0.348958
std        0.476951
min        0.000000
25%        0.000000
50%        0.000000
75%        1.000000
max        1.000000
Name: Outcome, dtype: float64

In [ ]:
Predominan los casos sin diagnóstico de diabetes, aunque existe una proporción relevante de positivos.

In [30]:
#Revisa Valores nulos con funcion isnull().sum()
df.isnull().sum()

Pregnancies                 0
Glucose                     0
BloodPressure               0
SkinThickness               0
Insulin                     0
BMI                         0
DiabetesPedigreeFunction    0
Age                         0
Outcome                     0
dtype: int64

In [ ]:
El conjunto de datos no presenta valores nulos, lo que garantiza su integridad para el análisis.

In [62]:
#Revisar valores únicos por columna usando función unique(): nombre-columna.unique()
df["Glucose"].unique()

array([148,  85, 183,  89, 137, 116,  78, 115, 197, 125, 110, 168, 139,
       189, 166, 100, 118, 107, 103, 126,  99, 196, 119, 143, 147,  97,
       145, 117, 109, 158,  88,  92, 122, 138, 102,  90, 111, 180, 133,
       106, 171, 159, 146,  71, 105, 101, 176, 150,  73, 187,  84,  44,
       141, 114,  95, 129,  79,   0,  62, 131, 112, 113,  74,  83, 136,
        80, 123,  81, 134, 142, 144,  93, 163, 151,  96, 155,  76, 160,
       124, 162, 132, 120, 173, 170, 128, 108, 154,  57, 156, 153, 188,
       152, 104,  87,  75, 179, 130, 194, 181, 135, 184, 140, 177, 164,
        91, 165,  86, 193, 191, 161, 167,  77, 182, 157, 178,  61,  98,
       127,  82,  72, 172,  94, 175, 195,  68, 186, 198, 121,  67, 174,
       199,  56, 169, 149,  65, 190])

In [ ]:
La variable Glucose presenta múltiples valores únicos, lo que refleja alta variabilidad entre los pacientes.

In [67]:
#Revisar valores únicos por columna usando función unique(): nombre-columna.unique()
df["BMI"].unique()

array([33.6, 26.6, 23.3, 28.1, 43.1, 25.6, 31. , 35.3, 30.5,  0. , 37.6,
       38. , 27.1, 30.1, 25.8, 30. , 45.8, 29.6, 43.3, 34.6, 39.3, 35.4,
       39.8, 29. , 36.6, 31.1, 39.4, 23.2, 22.2, 34.1, 36. , 31.6, 24.8,
       19.9, 27.6, 24. , 33.2, 32.9, 38.2, 37.1, 34. , 40.2, 22.7, 45.4,
       27.4, 42. , 29.7, 28. , 39.1, 19.4, 24.2, 24.4, 33.7, 34.7, 23. ,
       37.7, 46.8, 40.5, 41.5, 25. , 25.4, 32.8, 32.5, 42.7, 19.6, 28.9,
       28.6, 43.4, 35.1, 32. , 24.7, 32.6, 43.2, 22.4, 29.3, 24.6, 48.8,
       32.4, 38.5, 26.5, 19.1, 46.7, 23.8, 33.9, 20.4, 28.7, 49.7, 39. ,
       26.1, 22.5, 39.6, 29.5, 34.3, 37.4, 33.3, 31.2, 28.2, 53.2, 34.2,
       26.8, 55. , 42.9, 34.5, 27.9, 38.3, 21.1, 33.8, 30.8, 36.9, 39.5,
       27.3, 21.9, 40.6, 47.9, 50. , 25.2, 40.9, 37.2, 44.2, 29.9, 31.9,
       28.4, 43.5, 32.7, 67.1, 45. , 34.9, 27.7, 35.9, 22.6, 33.1, 30.4,
       52.3, 24.3, 22.9, 34.8, 30.9, 40.1, 23.9, 37.5, 35.5, 42.8, 42.6,
       41.8, 35.8, 37.8, 28.8, 23.6, 35.7, 36.7, 45

In [ ]:
El índice de masa corporal (BMI) presenta gran diversidad de valores, indicando variación en el estado físico de los participantes.

In [131]:
#Revisar valores únicos por columna usando función unique(): nombre-columna.unique()
df["Outcome"].unique()

array([181.6, 111.6, 206.3, 117.1, 180.1, 141.6, 109. , 150.3, 227.5,
       125. , 147.6, 206. , 166.1, 219.1, 191.8, 130. , 163.8, 136.6,
       146.3, 149.6, 165.3, 134.4, 235.8, 148. , 179.6, 156.1, 186.4,
       120.2, 167.2, 151.1, 145. , 189.6, 112.8, 111.9, 127. , 171.2,
       134.9, 128.2, 148.1, 214. , 173.2, 128.7, 216.4, 222. , 175.7,
        99. , 142.1, 105. , 122.4, 125.2, 112.4, 209.7, 184.7,  96. ,
       224.7, 146.8, 186.5, 146.5,  84. , 165.9,  69. , 166.4, 128. ,
       141.5, 151.7, 114.6, 174.9, 132.9, 167.6, 169.4, 164.1, 111. ,
        24.7,  94.6, 132.7, 174.2, 137. , 135.4,  74. , 112.3, 125.6,
       185.8, 142.4, 142.6, 138.5, 173.1, 133.5,  99.1, 155. , 127.7,
       157.8, 166.7, 177.9, 123.6,  91.4, 121.7, 171.7, 202. , 177.1,
       147.5, 107.6, 124.6, 154.7, 118.4, 173.5, 117.3, 132.4, 204.3,
       189. , 110. , 190.5, 177.2, 158. , 111.7, 122.2, 215.2, 145.2,
       140.6, 158.8, 143. , 162.9, 151.3, 151.5, 202.7, 155.3, 204.5,
       122.3, 130.8,

In [ ]:
La variable Outcome muestra alta variabilidad numérica, indicando diferencias significativas en los resultados de los pacientes.

In [68]:
df["Outcome"].unique()

array([1, 0])

## Variables Cuantitativas

### Medidas de tendencia central


In [43]:
#Glucose
#Se puede obtener la media, mediana y moda para
mean_Glucose = df['Glucose'].mean()
median_Glucose =df['Glucose'].median()
mode_Glucose = df['Glucose'].mode()
print("Mean_Glucose:",mean_Glucose)
print("Median_Glucose:",median_Glucose)
print("Mode_Glucose:",mode_Glucose)

Mean_Glucose: 120.89453125
Median_Glucose: 117.0
Mode_Glucose: 0     99
1    100
Name: Glucose, dtype: int64


Conclusiones:  
La Glucose promedio fue 121  
La Glucose al centro es 117 
La Glucose más repetida fue de 99



In [133]:
#BMI
#Se puede obtener la media, mediana y moda para
mean_BMI = df['BMI'].mean()
median_BMI =df['BMI'].median()
mode_BMI = df['BMI'].mode()
print("Mean_BMI:",mean_BMI)
print("Median_BMI:",median_BMI)
print("Mode_BMI:",mode_BMI)

Mean_BMI: 31.992578124999998
Median_BMI: 32.0
Mode_BMI: 0    32.0
Name: BMI, dtype: float64


Conclusiones:  
La BMI promedio fue 32  
La BMI al centro es 32  
La BMI más repetida fue de 32



In [132]:
#Outcome
#Se puede obtener la media, mediana y moda para
mean_Outcome = df['Outcome'].mean()
median_Outcome =df['Outcome'].median()
mode_Outcome = df['Outcome'].mode()
print("Mean_Outcome:",mean_Outcome)
print("Median_Outcome:",median_Outcome)
print("Mode_Outcome:",mode_Outcome)

Mean_Outcome: 152.887109375
Median_Outcome: 148.3
Mode_Outcome: 0     115.2
1     116.2
2     124.6
3     127.3
4     131.8
5     137.8
6     146.8
7     151.4
8     156.1
9     158.8
10    160.0
Name: Outcome, dtype: float64


Conclusiones:  
El Outcome promedio fue 153
El Outcome al centro es 148
El Outcome más repetida fue de 115


# Variables Categóricas

In [72]:
#Para conteo  de cada valor en una columna, en orden descendente usar función value_counts():    
# nombreDataframe.columna.value_counts()
# nombreDataframe['columna'].value_counts()
df.Glucose.value_counts()

Glucose
99     17
100    17
111    14
125    14
129    14
       ..
56      1
169     1
149     1
65      1
190     1
Name: count, Length: 136, dtype: int64

In [ ]:
Los valores de glucosa más comunes rondan entre 99 y 100, lo que indica una concentración frecuente cercana al límite normal.

In [108]:
#Para conteo  de cada valor en una columna, en orden descendente usar función value_counts():    
# nombreDataframe.columna.value_counts()
# nombreDataframe['columna'].value_counts()
df.BMI.value_counts()

BMI
32.0    13
31.6    12
31.2    12
0.0     11
32.4    10
        ..
49.6     1
24.1     1
41.2     1
49.3     1
46.3     1
Name: count, Length: 248, dtype: int64

In [ ]:
Los valores de BMI más frecuentes rondan entre 31 y 32, lo que sugiere una tendencia general hacia el sobrepeso en la muestra.

In [109]:
#Para conteo  de cada valor en una columna, en orden descendente usar función value_counts():    
# nombreDataframe.columna.value_counts()
# nombreDataframe['columna'].value_counts()
df.Outcome.value_counts()

Outcome
156.1    4
131.8    4
124.6    4
151.4    4
137.8    4
        ..
223.4    1
196.3    1
172.7    1
97.0     1
153.4    1
Name: count, Length: 551, dtype: int64

In [ ]:
Los valores más frecuentes del Outcome se concentran entre 124 y 156, reflejando una distribución equilibrada con pocos valores atípicos.

In [10]:
#Revisa conteo de varias columnas

In [75]:
df['BMI'].value_counts()

BMI
32.0    13
31.6    12
31.2    12
0.0     11
32.4    10
        ..
49.6     1
24.1     1
41.2     1
49.3     1
46.3     1
Name: count, Length: 248, dtype: int64

In [ ]:
El índice de masa corporal (BMI) más frecuente es de aproximadamente 32, indicando que la mayoría de los participantes presenta sobrepeso.

In [76]:
df['Glucose'].value_counts()

Glucose
99     17
100    17
111    14
125    14
129    14
       ..
56      1
169     1
149     1
65      1
190     1
Name: count, Length: 136, dtype: int64

In [ ]:
Los valores de glucosa más comunes se encuentran entre 99 y 125, indicando una tendencia general hacia niveles moderadamente elevados de glucosa.

In [111]:
df['Outcome'].value_counts()

Outcome
156.1    4
131.8    4
124.6    4
151.4    4
137.8    4
        ..
223.4    1
196.3    1
172.7    1
97.0     1
153.4    1
Name: count, Length: 551, dtype: int64

In [ ]:
Los valores del Outcome se concentran principalmente entre 124 y 156, mostrando una distribución homogénea con pocos valores extremos.

In [ ]:
# Crear variable totalScore que incluya la suma de las columnas Outcome, Glucose y BMI
# Mostrar los registros donde el total sea mayor o igual a 18
df["Outcome"] + df["Glucose"] + df["BMI"] >= 18

In [97]:
df

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50    181.6  
1                       0.351   31    111.6  
2                       0.672   32    206.3  
3                       0.167   21    117.1  
4                       2.288   33    180.1  
..                        ...  ...      ...  
763                     0.171   63    133.9  
764                     0.340   27    158.8  
765                     0.245   30    147.2  
766                     0.349   47    156.1  
767                     0.315   23    123.4  

[768 rows x 9 columns]

## Consulta

In [112]:
# df.iloc[i]: Accede a la fila en la posición i.
# Acceder a la primera fila
df.iloc[0]

Pregnancies                   6.000
Glucose                     148.000
BloodPressure                72.000
SkinThickness                35.000
Insulin                       0.000
BMI                          33.600
DiabetesPedigreeFunction      0.627
Age                          50.000
Outcome                     181.600
Name: 0, dtype: float64

In [ ]:
El primer registro muestra valores elevados de glucosa (148 ) y BMI (33.6), lo que indica una posible tendencia hacia el sobrepeso y un riesgo metabólico asociado.

In [99]:
# Acceder a las dos primeras filas
df.iloc[:2]

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50    181.6  
1                     0.351   31    111.6

In [ ]:
Las dos primeras filas muestran individuos con niveles distintos de glucosa y BMI, donde el primero presenta valores más altos, lo que sugiere un mayor riesgo metabólico en comparación con el segundo.

In [100]:
#Seleccionar columnas, indicando entre corchetes [nombreColumna, nombreColumna]
df[["Glucose", "BMI", "Outcome"]]

Glucose   BMI  Outcome
0        148  33.6    181.6
1         85  26.6    111.6
2        183  23.3    206.3
3         89  28.1    117.1
4        137  43.1    180.1
..       ...   ...      ...
763      101  32.9    133.9
764      122  36.8    158.8
765      121  26.2    147.2
766      126  30.1    156.1
767       93  30.4    123.4

[768 rows x 3 columns]

In [ ]:
Las columnas seleccionadas muestran que los valores de Glucose, BMI y Outcome varían significativamente entre los registros, evidenciando una posible relación entre niveles altos de glucosa, índice de masa corporal y el resultado clínico.

In [114]:
#Selección de filas [indicar dataframe[columna] operador valor]
df[df["Glucose"] >= 150]

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
2              8      183             64              0        0  23.3   
8              2      197             70             45      543  30.5   
11            10      168             74              0        0  38.0   
13             1      189             60             23      846  30.1   
14             5      166             72             19      175  25.8   
..           ...      ...            ...            ...      ...   ...   
749            6      162             62              0        0  24.3   
753            0      181             88             44      510  43.3   
754            8      154             78             32        0  32.4   
759            6      190             92              0        0  35.5   
761            9      170             74             31        0  44.0   

     DiabetesPedigreeFunction  Age  Outcome  
2                       0.672   32    206.3  
8                       0.158   53    227.5  
11                      0.537   34    206.0  
13                      0.398   59    219.1  
14                      0.587   51    191.8  
..                        ...  ...      ...  
749                     0.178   50    186.3  
753                     0.222   26    224.3  
754                     0.443   45    186.4  
759                     0.278   66    225.5  
761                     0.403   43    214.0  

[143 rows x 9 columns]

In [ ]:
Los datos seleccionados muestran que 143 registros presentan niveles de glucosa iguales o superiores a 150, lo que indica una alta proporción de individuos con riesgo potencial de hiperglucemia o diabetes.

In [115]:
#Selección de filas [indicar dataframe[columna] operador valor]
df[df["Outcome"] >= 150]

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
2              8      183             64              0        0  23.3   
4              0      137             40             35      168  43.1   
7             10      115              0              0        0  35.3   
8              2      197             70             45      543  30.5   
..           ...      ...            ...            ...      ...   ...   
757            0      123             72              0        0  36.3   
759            6      190             92              0        0  35.5   
761            9      170             74             31        0  44.0   
764            2      122             70             27        0  36.8   
766            1      126             60              0        0  30.1   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50    181.6  
2                       0.672   32    206.3  
4                       2.288   33    180.1  
7                       0.134   29    150.3  
8                       0.158   53    227.5  
..                        ...  ...      ...  
757                     0.258   52    159.3  
759                     0.278   66    225.5  
761                     0.403   43    214.0  
764                     0.340   27    158.8  
766                     0.349   47    156.1  

[371 rows x 9 columns]

In [ ]:
Se identificaron 371 registros con valores de Outcome mayores o iguales a 150, lo que sugiere que una parte considerable de la muestra presenta resultados clínicos elevados, posiblemente relacionados con un mayor riesgo metabólico o cardiovascular.

In [121]:
#Selección de filas [indicar dataframe[columna] operador valor]
df[df["BMI"] > 50]

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
120            0      162             76             56      100  53.2   
125            1       88             30             42       99  55.0   
177            0      129            110             46      130  67.1   
193           11      135              0              0        0  52.3   
247            0      165             90             33      680  52.3   
303            5      115             98              0        0  52.9   
445            0      180             78             63       14  59.4   
673            3      123            100             35      240  57.3   

     DiabetesPedigreeFunction  Age  Outcome  
120                     0.759   25    215.2  
125                     0.496   26    143.0  
177                     0.319   26    196.1  
193                     0.578   40    187.3  
247                     0.427   23    217.3  
303                     0.209   28    167.9  
445                     2.420   25    239.4  
673                     0.880   22    180.3

In [ ]:
Los registros con BMI mayor a 50 corresponden a individuos con obesidad severa. Estos casos, aunque poco frecuentes, reflejan un alto riesgo metabólico y una posible relación con valores elevados de glucosa y Outcome, lo que sugiere una mayor probabilidad de complicaciones relacionadas con la diabetes.

In [104]:
#ordenar usando funcion sort_values(by=atributo, ascending=True/false)
df.sort_values(by="BMI", ascending=True)

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
49             7      105              0              0        0   0.0   
60             2       84              0              0        0   0.0   
706           10      115              0              0        0   0.0   
81             2       74              0              0        0   0.0   
684            5      136             82              0        0   0.0   
..           ...      ...            ...            ...      ...   ...   
120            0      162             76             56      100  53.2   
125            1       88             30             42       99  55.0   
673            3      123            100             35      240  57.3   
445            0      180             78             63       14  59.4   
177            0      129            110             46      130  67.1   

     DiabetesPedigreeFunction  Age  Outcome  
49                      0.305   24    105.0  
60                      0.304   21     84.0  
706                     0.261   30    115.0  
81                      0.102   22     74.0  
684                     0.640   69    136.0  
..                        ...  ...      ...  
120                     0.759   25    215.2  
125                     0.496   26    143.0  
673                     0.880   22    180.3  
445                     2.420   25    239.4  
177                     0.319   26    196.1  

[768 rows x 9 columns]

In [122]:
#ordenar usando funcion sort_values(by=atributo, ascending=True/false)
df.sort_values(by="Glucose", ascending=True)

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
75             1        0             48             20        0  24.7   
349            5        0             80             32        0  41.0   
342            1        0             68             35        0  32.0   
502            6        0             68             41        0  39.0   
182            1        0             74             20       23  27.7   
..           ...      ...            ...            ...      ...   ...   
408            8      197             74              0        0  25.9   
579            2      197             70             99        0  34.7   
228            4      197             70             39      744  36.7   
561            0      198             66             32      274  41.3   
661            1      199             76             43        0  42.9   

     DiabetesPedigreeFunction  Age  Outcome  
75                      0.140   22     24.7  
349                     0.346   37     41.0  
342                     0.389   22     32.0  
502                     0.727   41     39.0  
182                     0.299   21     27.7  
..                        ...  ...      ...  
408                     1.191   39    222.9  
579                     0.575   62    231.7  
228                     2.329   31    233.7  
561                     0.502   28    239.3  
661                     1.394   22    241.9  

[768 rows x 9 columns]

In [123]:
#ordenar usando funcion sort_values(by=atributo, ascending=True/false)
df.sort_values(by="Outcome", ascending=True)

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
75             1        0             48             20        0  24.7   
182            1        0             74             20       23  27.7   
342            1        0             68             35        0  32.0   
502            6        0             68             41        0  39.0   
349            5        0             80             32        0  41.0   
..           ...      ...            ...            ...      ...   ...   
22             7      196             90              0        0  39.8   
154            8      188             78              0        0  47.9   
561            0      198             66             32      274  41.3   
445            0      180             78             63       14  59.4   
661            1      199             76             43        0  42.9   

     DiabetesPedigreeFunction  Age  Outcome  
75                      0.140   22     24.7  
182                     0.299   21     27.7  
342                     0.389   22     32.0  
502                     0.727   41     39.0  
349                     0.346   37     41.0  
..                        ...  ...      ...  
22                      0.451   41    235.8  
154                     0.137   43    235.9  
561                     0.502   28    239.3  
445                     2.420   25    239.4  
661                     1.394   22    241.9  

[768 rows x 9 columns]

In [105]:
#Agrupar por un atributo y calcular función de agregación utilizando groupby(atributo)['atributoAgregacion].funcion() (ej. mean() para promedio)
df.groupby("Outcome")[["Glucose", "BMI"]].mean()


Glucose   BMI
Outcome               
24.7         0.0  24.7
27.7         0.0  27.7
32.0         0.0  32.0
39.0         0.0  39.0
41.0         0.0  41.0
...          ...   ...
235.8      196.0  39.8
235.9      188.0  47.9
239.3      198.0  41.3
239.4      180.0  59.4
241.9      199.0  42.9

[551 rows x 2 columns]

Crea un subconjunto de diabetes para paciente con glucos mayor a 150

In [125]:
# Usa el criterio para extraer solo los pacientes con glucosa alta (Glucose > 150)
glucosa_alta = df[df["Glucose"] > 150]

In [126]:
df

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50    181.6  
1                       0.351   31    111.6  
2                       0.672   32    206.3  
3                       0.167   21    117.1  
4                       2.288   33    180.1  
..                        ...  ...      ...  
763                     0.171   63    133.9  
764                     0.340   27    158.8  
765                     0.245   30    147.2  
766                     0.349   47    156.1  
767                     0.315   23    123.4  

[768 rows x 9 columns]

In [ ]:
El filtro aplicado permite identificar a los pacientes con niveles de glucosa superiores a 150. Estos casos representan una población con alto riesgo de desarrollar diabetes o complicaciones metabólicas, siendo fundamental su monitoreo y análisis para estudios clínicos o preventivos.

Crea un subconjunto de diabetes para paciente con BMI Menor a 30

In [127]:
# Usa el criterio para extraer solo los pacientes con BMI baja (BMI > 30)
glucosa_alta = df[df["BMI"] > 30]

In [129]:
df

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50    181.6  
1                       0.351   31    111.6  
2                       0.672   32    206.3  
3                       0.167   21    117.1  
4                       2.288   33    180.1  
..                        ...  ...      ...  
763                     0.171   63    133.9  
764                     0.340   27    158.8  
765                     0.245   30    147.2  
766                     0.349   47    156.1  
767                     0.315   23    123.4  

[768 rows x 9 columns]

In [ ]:
El filtro se aplica para seleccionar pacientes con un índice de masa corporal (BMI) mayor a 30, es decir, aquellos que presentan sobrepeso u obesidad. Estos individuos suelen tener un mayor riesgo de resistencia a la insulina y complicaciones metabólicas, lo que los convierte en un grupo de especial interés para el análisis de la diabetes.

Crea un subconjunto de diabetes para el resultado con Outcome igual o mayor a 140

In [127]:
# Usa el criterio para extraer solo los resultado con 140 o mayor (Outcome > 140)
glucosa_alta = df[df["Outcome"] > 140]

In [130]:
df

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50    181.6  
1                       0.351   31    111.6  
2                       0.672   32    206.3  
3                       0.167   21    117.1  
4                       2.288   33    180.1  
..                        ...  ...      ...  
763                     0.171   63    133.9  
764                     0.340   27    158.8  
765                     0.245   30    147.2  
766                     0.349   47    156.1  
767                     0.315   23    123.4  

[768 rows x 9 columns]

In [ ]:
El filtro aplicado selecciona únicamente los registros con un Outcome superior a 140, lo que representa a los pacientes con resultados más elevados en el indicador de salud. Estos valores pueden asociarse a un mayor riesgo de complicaciones metabólicas o niveles anormales de glucosa, lo cual los convierte en casos relevantes para un análisis clínico detallado.